tests/unit/test_space.py

```diff
@@ -99,6 +122,51 @@ def test_discrete_search_space_sampling_raises_when_too_many_samples_are_request
         search_space.sample(num_samples)
 
 
+def test_discrete_search_space___mul___points_is_the_concatenation_of_original_points() -> None:
```

Comment:
Random thought: I remember we had a discussion about tests naming, but I don't remember what was the conclusion. I find it extremely difficult to read functions with long names, in the current case, wouldn't it better to have a short docstring for describing what function does instead of the 100 char function?

Reply:
i don't know. it's not really part of this PR though so do you want to make a ticket for it?

Reply:
my main concern with using docstrings for tests is that I can see them becoming an unmaintainable mess. It's pretty difficult keeping on top of docstrings for the source code